### DISCUSSION TITLE
有办法编译得更精简吗？用在MT7628/RAM:64MB/ROM:16MB

### DISCUSSION THREAD

**Discussion by kousyougi** [XTLS/Xray-core#667] [General]
我有一个路由器小米mini青春版：MT7628/RAM:64MB/ROM:16MB
ROM只有16MB装不下xray执行檔。
把xray用UPX压缩后放到ROM里面跑，会CPU满载，跑不了。
经过各种测试，如果用gzip解压缩到/tmp再执行。可以正常执行，以上网来说速度不慢。
但18MB的xray解压缩到/tmp几乎快把RAM占满了，有时RAM紧张时，整个系统会不太正常。

请问有没有什么方法可以compile更精简的xray。
我编译的openwrt只有4MB，只要能把xray执行檔压到10MB以下，让我不用解压缩到/tmp、直接放在ROM上执行，就能完美跑在这台路由器上了。

需要用的功能跟协议只有fakedns、tproxy、xtls。

[the remark this answers is not in the document]
**Reply by kousyougi** [reply]
还没把xray解压缩到/tmp时，系统的记忆体是这样：
Mem: 29660K used, 29312K free, 76K shrd, 2492K buff, 6416K cached

把18MB的xray复制到/tmp，没执行xray时，系统的记忆体是这样：
Mem: 47864K used, 11108K free, 17996K shrd, 2492K buff, 24340K cached

执行/tmp/xray之后系统记忆体是这样：
Mem: 49304K used, 9668K free, 17996K shrd, 1060K buff, 20284K cached

CPU负载不高，但free往下降时，SSH会卡卡的。但过一阵子可以恢复正常。

所以如果可以压在10MB以下，放在ROM执行，不解压缩复制到/tmp执行，可以节省内存。

目前测试是这样

[the remark this answers is not in the document]
**Reply by galaxyskyknight** [reply]
为啥要放到/tmp呢，/root不行吗

[the remark this answers is not in the document]
**Reply by kousyougi** [reply]
空间不够。
16MB的ROM，扣掉4MB的系统，大概只能装10MB的程序。xray目前编译出来是18MB。

**Comment by z826540272** [comment]
够用的,你自己的问题,不要用UPX,会占用20M内存.直接编译就行.

[the remark this answers is not in the document]
**Reply by z826540272** [reply]
你大概都没试过
